src/Validation/gen/ValidationsGenerator.cs

```diff
@@ -44,4 +47,34 @@ public void Initialize(IncrementalGeneratorInitializationContext context)
         // ValidatableTypeInfo for all validatable types.
         context.RegisterSourceOutput(emitInputs, Emit);
     }
+
+    private static void EmitValidatableTypeAttribute(IncrementalGeneratorPostInitializationContext context)
+    {
+        // First, emit the EmbeddedAttribute if it doesn't exist
+        var embeddedAttributeSource = """
+            // <auto-generated/>
+            namespace Microsoft.CodeAnalysis
+            {
+                [global::System.AttributeUsage(global::System.AttributeTargets.All, AllowMultiple = true, Inherited = false)]
+                internal sealed class EmbeddedAttribute : global::System.Attribute
+                {
+                }
+            }
+            """;
+
+        context.AddSource("EmbeddedAttribute.g.cs", embeddedAttributeSource);
```

Comment:
We don't have access to `AddEmbeddedAttributeDefinition` yet. We can refactor this once we get an updated compiler version?